docs/notary.md

```diff
@@ -0,0 +1,531 @@
+      to sign the request (and fallback transaction).
+    
+    `SignAndPushP2PNotaryRequest` will construct and sign fallback transaction,
+    construct and sign P2PNotaryRequest and submit it to the RPC node. The
+    resulting notary request and an error are returned.
+
+After P2PNotaryRequests are sent, participants should then wait for one of their
+transactions (main or fallback) to get accepted into one of subsequent blocks.
+
+### 3. Signatures collection and transaction release
+
+Valid P2PNotaryRequest payload is distributed via P2P network using standard
+broadcasting mechanisms until it reaches designated notary nodes that have the
+respective node module active. They collect all payloads for the same main
+transaction until enough signatures are collected to create proper witnesses for
+it. They then attach all witnesses required and send this transaction as usual
+and monitor subsequent blocks for its inclusion.
+
+All the operations leading to successful transaction creation are independent
+of the chain and could easily be done within one block interval, so if the
+first service request is sent at current height `H` it's highly likely that the
+main transaction will be a part of `H+1` block.
+ 
+### 4. Results monitoring
+
+Once P2PNotaryRequest reached RPC node, it is added to the notary request pool.
+Completed or outdated requests are being removed from the pool. Use
+[NeoGo notification subsystem](./notifications.md) to track request addition and
+removal:
+
+- Use RPC `subscribe` method with `notary_request_event` stream name parameter to
+  subscribe to `P2PNotaryRequest` payloads that are added or removed from the
+  notary request pool.
+- Use `sender` or `signer` filters to filter out notary request with desired
+  request senders or main tx signers.
+
+Use the notification subsystem to track that main or fallback transaction
+accepted to the chain:
+
+- Use RPC `subscribe` method with `transaction_added` stream name parameter to
+  subscribe to transactions that are accepted to the chain.
+- Use `sender` filter with Notary native contract hash to filter out fallback
+  transactions sent by Notary node. Use `signer` filter with notary request
+  sender address  to filter out fallback transactions sent by the specified
+  sender.
+- Use `sender` or `signer` filters to filter out main transaction with desired
+  sender or signers. You can also filter out main transaction using Notary
+  contract `signer` filter.
+- Don't rely on `sender` and `signer` filters only, check also that received
+  transaction has `NotaryAssisted` attribute with expected `NKeys` value.
+
+Use the notification subsystem to track main or fallback transaction execution
+results.
+
+Moreover, you can use all regular RPC calls to track main or fallback transaction
+invocation: `getrawtransaction`, `getapplicationlog` etc.
+
+## Notary service use-cases
+
+Here are presented several use-cases where Notary subsystem can be applied.
```

Comment:
The whole phrase feels strange to me. Maybe `Described here are ...` or `several .. are described below`?